src/Parsing/TreeSitter.hs

```diff
@@ -19,65 +20,71 @@ import           Data.Blob
 import           Data.Duration
 import           Data.Term
 import           Source.Loc
-import           Source.Source (Source)
 import qualified Source.Source as Source
 import           Source.Span
 
+import qualified TreeSitter.Cursor as TS
 import qualified TreeSitter.Language as TS
 import qualified TreeSitter.Node as TS
 import qualified TreeSitter.Parser as TS
 import qualified TreeSitter.Tree as TS
+import qualified TreeSitter.Unmarshal as TS
 
-data Result grammar
-  = Failed
-  | Succeeded (AST [] grammar)
```

Comment:
Removing this allowed me to share most of the setup &c for parsing between precise & à la carte ASTs.